crates/turborepo-repository/src/inference.rs

```diff
@@ -107,6 +107,23 @@ impl RepoState {
             .map(|root| root.into())
             .ok_or_else(|| Error::NotFound(reference_dir.to_owned()))
     }
+
+    pub fn get_mode(&self) -> RepoMode {
+        static WARN_INFERENCE: std::sync::OnceLock<()> = std::sync::OnceLock::new();
```

Comment:
This side-effect concerns me a bit since it means that if we change when we query the mode for the first time the warning will appear there instead. Also, if this happens to get called in the shim we could end up printing the warning twice.

Reply:
Originally this PR would print at calculation time but I was asked by Nathan to move it to query time instead. Perhaps we should just have manual check instead so we can control when this happens.